x-pack/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -1134,7 +1135,7 @@ export const tasks: TelemetryTask[] = [
       const response = await savedObjectsClient.find<SavedServiceGroup>({
         type: APM_SERVICE_GROUP_SAVED_OBJECT_TYPE,
         page: 1,
-        perPage: 50,
+        perPage: MAX_NUMBER_OF_SERVICE_GROUPS,
```

Comment:
It looks like this is actually retrieving the service groups instead of doing a count. Is that not possible with saved objects?

Reply:
Ah.. I see we retrieve the `kuery` field. Are we able to reduce the response from ES to just this field then? 
(Ideally we'd do a terms agg on `kuery` to avoid retrieving duplicates).

Reply:
Doing a terms agg with a size of 500 will also return more unique results (up to 500 to be exact) since it does not include duplicates.

Reply:
For the `kuery` I think we don't want to avoid duplicates. We want to collect all the possible `kuery_fields `

Reply:
What is the added value of two `kuery` strings that are the same? Are you counting the number of times a field is being used?

Reply:
One question though, does this collect telemetry for all spaces?

Reply:
Per the docs, I don't think so:

```
  namespaces?: string[];
  /**
   * This map defines each type to search for, and the namespace(s) to search for the type in; this is only intended to be used by a saved
   * object client wrapper.
   * If this is defined, it supersedes the `type` and `namespaces` fields when building the Elasticsearch query.
   * Any types that are not included in this map will be excluded entirely.
   * If a type is included but its value is undefined, the operation will search for that type in the Default namespace.
   */
```

Reply:
>  I don't understand why you don't want to avoid kuery duplicates

In order to see which fields are most often used.

> there's really no point in having service groups with the exact same kuery as others

that's right but there might be service groups that use same fields but different kuery. 

For example 2 service groups
```
{
  ...
  "groupName": "production java",
  "kuery": "service.environement: 'production' or agent.name:'java' ",
  ...
},
```

```
{
  ...
  "groupName": "stage envs",
  "kuery": "service.environement: 'stage' ",
  ...
},
```
we'll collect 

```
kuery_fields: ['service.environement', 'service.environement', 'agent.name',]
```

Reply:
@kpatticha Elasticsearch essentially deduplicates non-unique values anyway. I don't think there's a point in storing a single value multiple times.

But more importantly, can you make sure this telemetry is collected for _all_ spaces?

Reply:
oh! I wasn't aware about this. 

😱  that's good catch. 

`savedObjectsClient.find` searches in the default space. Let me fix it in a follow up

Reply:
follow up PR - https://github.com/elastic/kibana/pull/142856